estimagic/differentiation/hessian.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import pandas as pd
+
+from estimagic.differentiation.second_order_auxiliary import backward_hess
+from estimagic.differentiation.second_order_auxiliary import central_hess
+from estimagic.differentiation.second_order_auxiliary import forward_hess
+from estimagic.differentiation.second_order_auxiliary import hess_richardson
+
+
+def hessian(
+    func,
+    params_sr,
+    method="central",
+    extrapolant=None,
+    func_args=None,
+    func_kwargs=None,
+):
+    """
+    Calculate the hessian of *func*.
+
+    Args:
+        func (function): A function that maps params_sr into a float.
+        params_sr (Series): see :ref:`parmas_df`
+        func_args (list): additional positional arguments for func.
+        func_kwargs (dict): additional positional arguments for func.
+        extrapolant (string): A variable indicating the use of a extrapolation method.
+        method (string): The method for the computation of the derivative. Default is
+                         central as it gives the highest accuracy.
+
+    Returns:
+        DataFrame: The index and columns are the index of params_sr.
+
+    """
+    # set default arguments
+    func_args = [] if func_args is None else func_args
+    func_kwargs = {} if func_kwargs is None else func_kwargs
+    # The hessian matrix is indexed by the vatiable names
+    hess = pd.DataFrame(index=params_sr.index, columns=params_sr.index, dtype=float)
+    # Calculate the value of the function for the observations.
+    f_x0 = func(params_sr, *func_args, **func_kwargs)
+    # Check if a extrapolation method is given.
+    if not extrapolant:
+        # As in the 1st differentiation case, a seperate function is set for each
+        # method.
+        if method == "forward":
```

Comment:
you could use getattr instead of the if blocks to select the correct function. 

Also, f is not a very informative name, maybe you can find a better one.